container/client.go

```diff
@@ -111,7 +116,19 @@ func (client dockerClient) StartContainer(c Container) error {
 
 	log.Infof("Starting %s", name)
 
-	newContainerID, err := client.api.CreateContainer(config, name)
+	var err error
+	var newContainerID string
+	if username != "" && password != "" && email != "" {
+		auth := dockerclient.AuthConfig{
+			Username: username,
+			Password: password,
+			Email:    email,
+		}
+		newContainerID, err = client.api.CreateContainer(config, name, &auth)
```

Comment:
Maybe set `dockerclient.URL` manually to `os.Getenv("REPO_REGISTRY")`, if not empty, or something?